exponential_backoff.go

```diff
@@ -0,0 +1,60 @@
+package frankenphp
+
+import (
+	"sync"
+	"time"
+)
+
+const maxBackoff = 1 * time.Second
+const minBackoff = 100 * time.Millisecond
+const maxConsecutiveFailures = 6
+
+type exponentialBackoff struct {
+	backoff      time.Duration
+	failureCount int
+	mu           sync.RWMutex
+	upFunc       sync.Once
+}
+
+func newExponentialBackoff() *exponentialBackoff {
+	return &exponentialBackoff{backoff: minBackoff}
+}
+
+func (e *exponentialBackoff) reset() {
```

Comment:
Since this is in the `frankenphp` package, might be good to come up with less generic names?